### PR TITLE
CXP-1212: Create front component catalog list

### PR DESCRIPTION
**Pull request by masacc**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/front/src/components/CatalogList/components/List.test.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import {render, screen} from '@testing-library/react';
+import {ThemeProvider} from 'styled-components';
+import {pimTheme} from 'akeneo-design-system';
+import {List} from './List';
+import {useCatalogs} from '../hooks/useCatalogs';
+
+test('it renders without error', () => {
+    (useCatalogs as jest.Mock).mockReturnValue({
+        isLoading: false,
+        isError: false,
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/CatalogList/components/List.test.tsx:13]: add error field

File: components/catalogs/front/src/components/CatalogList/components/List.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                <Table.Header>
+                    <Table.HeaderCell>{translate('akeneo_catalogs.catalog_list.catalogs_name')}</Table.HeaderCell>
+                    <Table.HeaderCell>{translate('akeneo_catalogs.catalog_list.status')}</Table.HeaderCell>
+                </Table.Header>
+                <Table.Body>
+                    {catalogs.data?.map(catalog => (
+                        <Table.Row key={catalog.id}>
+                            <Table.Cell>{catalog.name}</Table.Cell>
+                            <Table.Cell>
+                                {catalog.enabled ? (
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/CatalogList/components/List.tsx:36]: transform into Status component

File: components/catalogs/front/src/components/CatalogList/components/List.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+const List: FC<PropsWithChildren<Props>> = ({owner}) => {
+    const translate = useTranslate();
+
+    const catalogs = useCatalogs(owner);
+
+    if (catalogs.isLoading) {
+        return null;
+    }
+
+    if (catalogs.isError) {
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/CatalogList/components/List.tsx:20]:
```suggestion
    if (catalogs.isError || undefined === catalogs.data) {
```

File: components/catalogs/front/src/components/CatalogList/components/List.test.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    (console.error as jest.Mock).mockImplementation(() => {});
+
+    expect(() => {
+        render(
+            <ThemeProvider theme={pimTheme}>
+                <List owner={'username'} />
+            </ThemeProvider>
+        );
+    }).toThrow(Error);
+});
```
Comment (guillaumefer-akeneo) [inline components/catalogs/front/src/components/CatalogList/components/List.test.tsx:82]: add a test when catalogs.data === undefined

File: components/catalogs/front/src/components/CatalogList/CatalogList.test.tsx [hunk 5]
```diff
@@ -1,17 +1,21 @@
...
 test('it renders without error', () => {
     render(
         <ThemeProvider theme={pimTheme}>
-            <CatalogList />
+            <CatalogList owner={'willy'} />
         </ThemeProvider>
     );
 
-    expect(screen.getByText('CatalogList')).toBeInTheDocument();
+    expect(screen.getByText('[list willy]')).toBeInTheDocument();
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogList/CatalogList.test.tsx:20]:
```suggestion
    expect(screen.getByText('[List]')).toBeInTheDocument();
```

File: components/catalogs/front/src/components/CatalogList/components/__mocks__/List.tsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import React, {FC, PropsWithChildren} from 'react';
+
+type Props = {
+    owner: string;
+};
+
+const List: FC<PropsWithChildren<Props>> = ({owner}) => <>[list {owner}]</>;
```
Comment (tseho) [inline components/catalogs/front/src/components/CatalogList/components/__mocks__/List.tsx:7]:
```suggestion
const List: FC = () => <>[List]</>;
```

**Review by tseho — APPROVED** [review]

**Review by guillaumefer-akeneo — APPROVED** [review]

**Pull request merged**